core/client/src/main/java/alluxio/client/file/options/CompleteUfsFileOptions.java

```diff
@@ -11,53 +11,50 @@
 
 package alluxio.client.file.options;
 
+import alluxio.Constants;
 import alluxio.annotation.PublicApi;
 import alluxio.client.ClientContext;
-import alluxio.security.LoginUser;
+import alluxio.security.authorization.Permission;
 import alluxio.thrift.CompleteUfsFileTOptions;
-import alluxio.util.SecurityUtils;
 
 import com.google.common.base.Objects;
 
+import java.io.IOException;
+
 import javax.annotation.concurrent.NotThreadSafe;
 
 /**
- * Options for completing a UFS file. Currently we do not allow users to set arbitrary user and
- * group options. The user and group will be set to the user login.
+ * Options for completing a UFS file. Currently we do not allow users to set arbitrary owner and
+ * group options. The owner and group will be set to the user login.
  */
 @PublicApi
 @NotThreadSafe
 public final class CompleteUfsFileOptions {
-  /** The ufs user this file should be owned by. */
-  private final String mOwner;
+  /** The ufs owner this file should be owned by. */
+  private String mOwner;
   /** The ufs group this file should be owned by. */
-  private final String mGroup;
+  private String mGroup;
+  /** The ufs permission in short format, e.g. 0777. */
+  private short mMode;
 
   /**
    * @return the default {@link CompleteUfsFileOptions}
+   * @throws IOException if failed to set owner from login module
    */
-  //TODO(calvin): Move the user group discovery to a utils method
-  //TODO(calvin): Utilize user-to-group mapping to discover the user
-  public static CompleteUfsFileOptions defaults() {
-    String user = null;
-    if (SecurityUtils.isAuthenticationEnabled(ClientContext.getConf())) {
-      try {
-        user = LoginUser.get(ClientContext.getConf()).getName();
-      } catch (Exception e) {
-        // Fall through to system property approach
-      }
-    }
-    if (user == null) { // Set to system user, or null if there is no system user set
-      String systemUser = System.getProperty("user.name");
-      user = systemUser.isEmpty() ? null : systemUser;
-    }
-    String group = user;
-    return new CompleteUfsFileOptions(user, group);
+  public static CompleteUfsFileOptions defaults() throws IOException {
```

Comment:
since the constructor here is nontrivial, better to javadoc what is the default here.
basically, telling people this default option value will contain owner group and mode from login module.